perf.proto

```diff
@@ -89,23 +90,42 @@ message MetricsCounters {
 
 message MetricsTimers {
   google.protobuf.Duration duration = 1;
-  google.protobuf.Duration waiting = 2;
+  google.protobuf.Duration total = 2;
 }
 
-message MetricsState {
-  bool failed = 1;
+message MetricsGuages {
+  int64 state = 1;
   int64 workers = 2;
+  bool failed = 3;
 }
 
 message MetricsEvent {
   string id = 1;
   repeated MetricsPoint Event = 3;
 }
 
+message RollupValue {
+  string name = 1;
+  oneof value {
+    int64 int = 2;
+    float fl = 3;
+  }
+  int64 version = 4;
+}
+
+message Rollups {
+  repeated RollupValue default_stats = 1;
+  repeated RollupValue user_stats = 2;
+  google.protobuf.Timestamp processed_at = 3;
+  int64 count = 4;
+  bool valid = 5;
+}
+
 service SinkPerformanceMetrics {
   rpc CreateMetricSeries(ResultData) returns (MetricsResponse);
   rpc AttachResultData(ResultData) returns (MetricsResponse);
   rpc AttachAuxilaryData(ResultData) returns (MetricsResponse);
+  rpc AttachRollups(ResultData) returns (MetricsResponse);
```

Comment:
maybe we want a message type that just wraps rollups (so you can associate an ID with it.) ? Otherwise everything takes the same type?

Reply:
So I was just following the signatures of the rest of these functions, if we do that then shouldn't we also do it for AttachAuxilaryData?

Reply:
yes probably.

Reply:
If we do this then we expect the user to have access to the ID (which is created by hashing and combining the fields in PerformanceResultInfo). Is that expected behavior?

Reply:
Yeah, because it's included in the CreateMetricSeries response.